include/dpp/emoji.h

```diff
@@ -55,38 +55,66 @@ class DPP_EXPORT emoji : public managed, public json_interface<emoji>  {
 	/**
 	 * @brief Emoji name
 	 */
-	std::string name;
+	std::string name{};
 	/**
 	 * @brief User id who uploaded the emoji
 	 */
-	snowflake user_id;
+	snowflake user_id{0};
 	/**
 	 * @brief Flags for the emoji from dpp::emoji_flags
 	 */
-	uint8_t flags;
+	uint8_t flags{0};
 	/**
 	 * @brief Image data for the emoji if uploading
 	 */
-	std::string* image_data;
-	
+	std::string image_data{};
```

Comment:
the reason we didn't have this as string is that a string is 12 bytes and a pointer is 8 or 4. this is only set when you are uploading an emoji, but, there are millions of emojis cached in a running bot at scale. this adds megabytes to the bot. can you possibly rethink this to not have a pointless empty std::string on every emoji instance?

Reply:
Do you suggest removing this field entirely and making it a parameter on the request functions? Would help with the memory rather more than having pointers in each object.

Reply:
I do like that idea